conditional/util/ldap.py

```diff
@@ -1,250 +1,120 @@
+def ldap_get_intro_members():
+    return _ldap_get_group_members("intromembers")
 
 
 @lru_cache(maxsize=1024)
 def ldap_get_onfloor_members():
-    return [x for x in ldap_get_current_students()
-            if ldap_is_onfloor(x['uid'][0].decode('utf-8'))]
+    return _ldap_get_group_members("onfloor")
 
 
 @lru_cache(maxsize=1024)
 def ldap_get_current_students():
-    return [x[1]
-            for x in _ldap_get_members()[1:]
-            if ldap_is_current_student(str(str(x[0]).split(",")[0]).split("=")[1])]
+    return _ldap_get_group_members("current_student")
 
 
-def ldap_is_active(username):
-    return _ldap_is_member_of_group(username, 'active')
+def ldap_is_active(account):
+    return _ldap_is_member_of_group(account, 'active')
 
 
-def ldap_is_alumni(username):
-    # When alumni status becomes a group rather than an attribute this will
-    # change to use _ldap_is_member_of_group.
-    alum_status = _ldap_get_field(username, 'alumni')
-    return alum_status is not None and alum_status.decode('utf-8') == '1'
+def ldap_is_alumni(account):
+    # If the user is not active, they are an alumni.
+    return not _ldap_is_member_of_group(account, 'active')
 
 
-def ldap_is_eboard(username):
-    return _ldap_is_member_of_group(username, 'eboard')
+def ldap_is_eboard(account):
+    return _ldap_is_member_of_group(account, 'eboard')
 
 
-def ldap_is_rtp(username):
-    return _ldap_is_member_of_group(username, 'rtp')
+def ldap_is_rtp(account):
+    return _ldap_is_member_of_group(account, 'rtp')
 
 
-def ldap_is_intromember(username):
-    return _ldap_is_member_of_group(username, 'intromembers')
+def ldap_is_intromember(account):
+    return _ldap_is_member_of_group(account, 'intromembers')
 
 
-def ldap_is_onfloor(username):
-    # april 3rd created onfloor group
-    # onfloor_status = _ldap_get_field(username, 'onfloor')
-    # return onfloor_status != None and onfloor_status.decode('utf-8') == '1'
-    return _ldap_is_member_of_group(username, 'onfloor')
+def ldap_is_onfloor(account):
+    return _ldap_is_member_of_group(account, 'onfloor')
 
 
-def ldap_is_financial_director(username):
-    return _ldap_is_member_of_committee(username, 'Financial')
+def ldap_is_financial_director(account):
+    return _ldap_is_member_of_committee(account, 'Financial')
 
 
-def ldap_is_eval_director(username):
+def ldap_is_eval_director(account):
     # TODO FIXME Evaulations -> Evaluations
-    return _ldap_is_member_of_committee(username, 'Evaulations')
+    return _ldap_is_member_of_committee(account, 'Evaulations')
 
 
-def ldap_is_current_student(username):
-    return _ldap_is_member_of_group(username, 'current_student')
+def ldap_is_current_student(account):
+    return _ldap_is_member_of_group(account, 'current_student')
 
 
-def ldap_set_housingpoints(username, housing_points):
-    _ldap_set_field(username, 'housingPoints', housing_points)
+def ldap_set_housingpoints(account, housing_points):
+    account.housingPoints = housing_points
```

Comment:
We should actually check if `config['LDAP_RO']` is set here (and for the rest of the write functions) instead of ignoring it.